lib/src/view/broadcast/broadcast_boards_tab.dart

```diff
@@ -159,7 +159,7 @@ class BroadcastPreview extends ConsumerWidget {
           final playingSide = Setup.parseFen(game.fen).turn;
 
           return ObservedBoardThumbnail(
-            boardKey: Key('Board-$index'),
+            boardKey: ValueKey(game.id),
```

Comment:
It should be the concatenation of `game.id` and `board-$index`. The key must be unique within the array as well.